ReleaseProcessScript/tools/Core/git_base_functions.ps1

```diff
@@ -22,14 +22,36 @@ function Get-Current-Branchname ()
   return $(git symbolic-ref --short -q HEAD)
 }
 
-function Get-Last-Version-Of-Branch-From-Tag-Exists ($Branchname)
+function Get-Last-Version-Of-Branch-From-Tag-Exists ($From, $To)
```

Comment:
Could you rename the parameters to make it more explicit what `$From` and `$To` are? Are they tags names, branch names, hashes etc.

Reply:
I am no expert on git terminology, so I might need help from someone who is on that one.

`$From` and `$To` are used to denote revisions. They can be tag names (implying the commit the tag points to - I only tested it with annotated tags, though, would have to test lightweight tags as well), branch names (referring to the tip of the branch) or commit hashes (self-explanatory).

The documentation describes the parameter as _object_ and/or commit:

> --merged[=\<object>]
Only list refs whose tips are reachable from the specified commit (HEAD if not specified).

So in the end it is a commit. One might say the parameters accept `commit-ish`, as defined by the official documentation:

> commit-ish (also committish)
A commit object or an object that can be recursively dereferenced to a commit object. The following are all commit-ishes: a commit object, a tag object that points to a commit object, a tag object that points to a tag object that points to a commit object, etc.

(source: https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefcommit-ishacommit-ishalsocommittish)

I am not sure if, by that definition, the name of a branch is also a `commit-ish` - in any case, it definitely works as parameter for this command.

So would `$FromCommitish` and `$ToCommitish` be more to your liking? I personally wouldn't fancy it, but if you think it's more descriptive, I'll gladly change it - the same goes for any other (better) suggestion you might have.

Reply:
I agree that `$FromCommitish` sounds kind of dumb, but it **is** the official terminology. The information you collected in your answeris very good, thank you very much for it. Considering your opinion, may I suggest adding the [powershell equivalent of javadoc](https://docs.microsoft.com/en-us/powershell/scripting/developer/help/examples-of-comment-based-help?view=powershell-7.1) to the functions? I genuinely feel like we have to _somehow_ add the information what `$To` and `$From` actually are, maybe it's changing the parameter name, maybe it's using powershell's comment-based-help or maybe it's just adding plain comments underneath the signature. You could try adding the comment-based-help thingies and see if it suits the function/file and looks any good. Otherwise, maybe just 2 plain comment lines briefly explaining what a commit-ish object is would suffice, I will leave it up to you (and finally Michael) to decide what's best

Reply:
I completely agree with you that it's best to include information on what `$To` and `$From` really stand for. I have updated the respective file with "ordinary" comments next to the formal arguments. I actually considered the Comment-Based Help you linked before opening the PR but opted against it because this function/these functions would be "the odd ones out" in the greater scope of the release process script as a whole (such extensive documentation is only found in the main entry points in `ReleaseProcessScript.psm1`). However, one could argue that it's worth adding such extensive documentation, i. e. moving comments from function body to help text and possibly even adding one or two examples in this help text. This should, in my opinion, only apply to this function, though - for the other ones you commented on below, short comments next next to the parameter definition should suffice (to me at least, since I do not want to spam the file with repetetive documentation).

But I agree that, of course, Michael has the final say in what's the best route to go.